### PR TITLE
Start API 0.2: page subtitles, timezone, disallow duplicate keys

### PR DESCRIPTION
**Pull request by encukou**
This provides the `naucse_render` side fixes for:
- https://github.com/pyvec/naucse/issues/20 – Breadcrumbs don't respect subpages
- https://github.com/pyvec/naucse/issues/13 – API: Time information lacks timezone info
- https://github.com/pyvec/naucse/issues/2 – Prevent duplicate slugs
- (partially) https://github.com/pyvec/naucse_render/issues/7 – Automatically convert ./linux to {{ subpage_url('linux') }}

For most of these, `naucse` will also need an update to use the new information.

### CODE REVIEW

**Comment by encukou** [comment]
Please leave a comment if you plan to review this.

**Comment by encukou** [comment]
@hroncok, do you want to look at this one?

**Comment by hroncok** [comment]
This has slipped from my radar. I will have a look.

**Comment by encukou** [comment]
Any ETA? This is starting to block me :(

**Comment by hroncok** [comment]
No ETA. Sorry about this, will prioritize. I have hundreds other things to look at. Is it starting to block you in matter of hours, days or weeks?

A.k.a do i need to stop everything and look now, does it wait for the afternoon, does it way for e.g. Friday?

**Comment by encukou** [comment]
Friday is OK, but if you can find time early afternoon, it would be great if you can have a quick look for potential problems (here and in the related https://github.com/pyvec/naucse/pull/29).
I plan to work on naucse on Mondays. Time between courses is great for the infrastructure; there's about a month and a half left before the next course starts.

**Comment by hroncok** [comment]
OK. Feel free to ping me more often about this. It seems to be one of the cursed tasks (the one I unconsciously tend to forgot about).

File: naucse_render/lesson.py [hunk 1]
```diff
@@ -56,10 +56,15 @@ def get_lesson(lesson_slug, vars, base_path):
...
-
+    pages_info.setdefault('index', {'title': lesson_info['title']})
     for slug, page_info in pages_info.items():
-        info = {**lesson_info, **page_info}
+        info = {**lesson_info, 'title': None, **page_info}
+        if 'title' not in page_info:
+            try:
+                subtitle = info['subtitle']
+            except KeyError:
+                raise ValueError(f"'subtitle' is required for page {lesson_slug}/{slug}")
```
Comment (hroncok) [inline naucse_render/lesson.py]: subtitle or title?

File: test_naucse_render/test_load.py [hunk 2]
```diff
@@ -73,3 +74,38 @@ def test_isolated_load(tmp_path):
...
+@pytest.mark.xfail(
+    strict=True,
+    reason="Incomplete workaround for https://github.com/yaml/pyyaml/issues/165"
+)
+def test_read_yaml_allow_merge(tmp_path):
+    """Assert that read_yaml allows overriding dict keys in YAML merge"""
+    yaml_path = tmp_path / 'test.yaml'
+    yaml_path.write_text("""data:
+        1:
+            <<: &common
```
Comment (hroncok) [inline test_naucse_render/test_load.py]: Use YAML, they said. It is easy -- like plaintext, they said.

**Comment by frenzymadness** [comment]
> there's about a month and a half left before the next course starts.

Two courses in Ostrava start in 3 weeks, PyData in Prague in a week and half.

File: test_naucse_render/test_load.py (hunk 2 above)
Comment (encukou) [reply test_naucse_render/test_load.py]: Well, YAML is the Ruby of markup languages :)
But this particular feature doesn't bother me, actually.

File: naucse_render/lesson.py (hunk 1 above)
Comment (encukou) [reply naucse_render/lesson.py]: This is for subpages (not the index), where I'd strongly prefer `subtitle`.
Adding `title` will avoid the error. Removing the subpage will also do. But you should really only add `title` if you know what you're doing.

**Comment by encukou** [comment]
@frenzymadness 
> Two courses in Ostrava start in 3 weeks, PyData in Prague in a week and half.

Right – please ping me after DevConf if there's still anything that needs to be released.
My comment was more about the time when *I* will be able to focus on infrastructure rather than preparing content :)

File: naucse_render/lesson.py (hunk 1 above)
Comment (hroncok) [reply naucse_render/lesson.py]: ack

**Comment by encukou** [comment]
Thank you! I'll merge now, and if we have time, let's test. And fix any bugs in future commits.

**Pull request merged**